optimum/exporters/onnx/__main__.py

```diff
@@ -116,7 +116,12 @@ def main():
         args.for_ort and (model.config.is_encoder_decoder or task.startswith("causal-lm"))
     ):
         if task == "stable-diffusion":
-            output_names = ["text_encoder/model.onnx", "unet/model.onnx", "vae_decoder/model.onnx"]
+            output_names = [
+                "text_encoder/model.onnx",
+                "unet/model.onnx",
+                "vae_encoder/model.onnx",
```

Comment:
Is the encoder always necessary? For example `pipeline_stable_diffusion.py` never seems to call `encode()`.

Reply:
No you're right it depends on the task but in my opinion it makes sense to export the entire model which can be later used for any tasks